pkg/process/checks/discovery_check.go

```diff
@@ -0,0 +1,63 @@
+package checks
+
+import (
+	"os"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var Discovery = &DiscoveryCheck{}
+
+type DiscoveryCheck struct {
+	probe *procutil.Probe
+	info  *model.SystemInfo
+}
+
+func (d *DiscoveryCheck) Init(cfg *config.AgentConfig, info *model.SystemInfo) {
+	d.probe = procutil.NewProcessProbe()
+	d.info = info
+}
+
+func (d *DiscoveryCheck) Name() string { return "discovery" }
+
+func (d *DiscoveryCheck) RealTime() bool { return false }
+
+func (d *DiscoveryCheck) Run(_ *config.AgentConfig, groupID int32) ([]model.MessageBody, error) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		_ = log.Warn("unable to get hostname")
+		hostname = "unknown"
+	}
+
+	procs, err := getAllProcesses(d.probe)
```

Comment:
Cool, yeah, this could be a great start for what we need but we probably need a comment to highlight the fact that we don't need any process stats collected, just the metadata (collecting stats can be expensive).

Reply:
Alright cool, I can create a card to add a `MetadataOnly` option to the process collector

Reply:
It should be really done as part of this PR.

Reply:
Alright, I can go ahead and put it in this pr

Reply:
Should be fixed now!